### ISSUE TITLE
CLI?

### ISSUE THREAD

**Issue by alexreg** [DotNetAnalyzers/StyleCopAnalyzers#2571]
Does StyleCop.Analyzers have a command-line interface? This would be especially useful on Mono, where there is no code analysis integration in Xamarin Studio.

**Comment by llehn** [comment]
Used as a nuget package, it integrates into the build. No CLI needed.

**Comment by alexreg** [comment]
You say "needed", but actually a CLI is sometimes desirable over integration... in this case though, I have since learnt that you cannot do the static analysis without building (which is rather odd)... so I guess this is a no-go.

**Comment by llehn** [comment]
Well its a roslyn based analyzer, it has to build to do its work, since roslyn is a compiler.

Out of curiousity, in what scenario would a CLI be more useful than an integration?

**Comment by alexreg** [comment]
Need not perform every stage of the build though, I was thinking.

Would be more useful for automation, e.g. CI. With some MSBuild parameters though (I have a separate issue on that), this issue could be alleviated.

**Comment by suras48** [comment]
alex,
even  i have the same requirement, so i have used StyleCopCmd.exe which can be used in command line.
it works fine.

But is der any other tool which is available in Nuget any idea?
Have solved this?

**Comment by andreacassioli** [comment]
I agree having a proper CLI would be terrific, especially when it comes to CI.

**Comment by andreacassioli** [comment]
Having a CLI utility would be amazing to work both with CI but also as git commit hook.

**Comment by w1zeman1p** [comment]
Is this still true? I'm helping maintain client libraries / bindings in many different languages: ruby, python, js, java, kotlin, and c#. As part of this project, I'm shelling out to various language specific cli tools for linting snippets of code for documentation. Would love any pointers for running StyleCop against my c# snippets from the CLI (mac/unix)

**Comment by andreacassioli** [comment]
Hey maintainers, any thoughts about this issue after years of silence?

**Comment by elachlan** [comment]
I found this but didn't get around to trying it. 

https://github.com/kzu/AutoCodeFix
